eng/scripts/Armstrong-Validation.ps1

```diff
@@ -0,0 +1,200 @@
+
+  if ($script:psYamlInstalled) {
+    LogInfo "Module powershell-yaml@0.4.7 already installed"
+  }
+  else {
+    LogInfo "Installing module powershell-yaml@0.4.7"
+    Install-Module -Name powershell-yaml -RequiredVersion 0.4.7 -Force -Scope CurrentUser
+    $script:psYamlInstalled = $true
+  }
+}
+
+function Find-Suppressions-Yaml {
+  param (
+    [string]$fileInSpecFolder
+  )
+
+  $currentDirectory = Get-Item (Split-Path -Path $fileInSpecFolder)
+
+  while ($currentDirectory) {
+    $suppressionsFile = Join-Path -Path $currentDirectory.FullName -ChildPath "suppressions.yaml"
+
+    if (Test-Path $suppressionsFile) {
+      return $suppressionsFile
+    }
+    else {
+      $currentDirectory = $currentDirectory.Parent
+    }
+  }
+
+  return $null
+}
+
+function Get-Suppression {
+  param (
+    [string]$fileInSpecFolder
+  )
+
+  $suppressionsFile = Find-Suppressions-Yaml $fileInSpecFolder
+  if ($suppressionsFile) {
+    Ensure-PowerShell-Yaml-Installed
+
+    $suppressions = Get-Content -Path $suppressionsFile -Raw | ConvertFrom-Yaml
+    foreach ($suppression in $suppressions) {
+      $tool = $suppression["tool"]
+      $path = $suppression["path"]
+
+      if ($tool -eq "ArmstrongValidation") {
+        # Paths in suppressions.yml are relative to the file itself
+        $fullPath = Join-Path -Path (Split-Path -Path $suppressionsFile) -ChildPath $path
+
+        # If path is not specified, suppression applies to all files
+        if (!$path -or ($fileInSpecFolder -like $fullPath)) {
+          return $suppression
+        }
+      }
+    }
+  }
+
+  return $null
+}
+
+function Get-ChangedTerraformFiles($changedFiles = (Get-ChangedFiles)) {
+  $changedFiles = Get-ChangedFilesUnderSpecification $changedFiles
+
+  $changedSwaggerFiles = $changedFiles.Where({ 
+      $_.EndsWith("main.tf")
+    })
+    
+  return $changedSwaggerFiles
+}
+
+$script:armstrongInstalled = $false
+function Ensure-Armstrong-Installed {
+  if ($script:armstrongInstalled) {
+    # If already checked once in this script, don't log anything further
+    return;
+  }
+
+  $script:armstrongInstalled = $true
+
+  if ($PSVersionTable.OS -like "*Windows*") {
+    LogInfo "Skipping Armstrong install on Windows"
+    return;
+  }
+
+  # Armstrong will be installed under $HOME/go/bin
+  $armstrongPath = Join-Path -Path $HOME -ChildPath "go/bin"
+  $env:PATH += ":$armstrongPath"
+
+  # install golang
+  if (!(Get-Command "go" -ErrorAction SilentlyContinue)) {
+    LogInfo "Installing Go..."
+    sudo apt-get update
+    sudo apt-get install -y golang-go
+  }
+  else {
+    LogInfo "Go is already installed."
+  }
+
+
+  # install armstrong
+  if (!(Get-Command "armstrong" -ErrorAction SilentlyContinue)) {
+    LogInfo "Installing Armstrong..."
+    go install github.com/azure/armstrong@latest
+  }
+  else {
+    LogInfo "Armstrong is already installed."
+  }
```

Comment:
For the PR check, you should move this code into `armstrong-validation.yml`, and make the following changes:

1. Install Go using the DevOps task, rather than directly calling `apt-get install`: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/go-tool-v0?view=azure-pipelines
2. Pin dependency `github.com/azure/armstrong@latest` to a specific version or commit SHA.  For repeatable builds (same behavior from a given commit SHA), all dependencies should be pinned.  We should never use `latest` which can change and break at any time.

Reply:
Thank you for the advice. I have moved the logic to .yml and use the fixed commit SHA for armstrong in the new PR.